test/shared/src/main/scala/zio/test/DefaultTestReporter.scala

```diff
@@ -38,10 +38,12 @@ object DefaultTestReporter {
                          case Spec.TestCase(_, test) => test.map(_.isLeft);
                          case _                      => UIO.succeed(false)
                        })
-            hasFailures   = failures.exists(identity)
-            status        = if (hasFailures) Failed else Passed
-            renderedLabel = if (hasFailures) renderFailureLabel(label, depth) else renderSuccessLabel(label, depth)
-            rest          <- UIO.foreach(specs)(loop(_, depth + tabSize)).map(_.flatten)
+            hasFailures = failures.exists(identity)
+            status      = if (hasFailures) Failed else Passed
+            renderedLabel = if (specs.isEmpty) ""
```

Comment:
Is it possible for `specs` to be empty? Aren't we defaulting to running all the tests if the filter predicate isn't matching any tests? I don't think we need to make any changes to this file, though I could be wrong.

Reply:
This addition isn't really for when the root Suite's specs are empty. It's more for the nested Suites below that. If we don't have this, then you get output like:

```
+ ZScheduleSpec
  + Repeat on success according to a provided strategy
  + Collect all inputs into a list
  + Repeat an action 2 times and call `ensuring` should
  + Retry on failure according to a provided strategy
    + retry 0 time for `once` when first time succeeds
  + Retry according to a provided strategy
  + Return the result of the fallback after failing and no more retries left
  + Return the result after successful retry
  + Retry a failed action 2 times and call `ensuring` should
```

as opposed to:

```
+ ZScheduleSpec
  + Retry on failure according to a provided strategy
    + retry 0 time for `once` when first time succeeds
```

I much prefer the latter (only show the selected nodes). Well, you could color the ignored specs gray or something, but it's still a lot of noise in my opinion. Especially for large specs where you would have to scroll to see the part you care about.

Reply:
Got it. That makes sense.